openupgrade_scripts/scripts/website_sale/14.0.1.0/post-migration-website_snippet_carousel_product.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2022 Tecnativa - David Vidal
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from lxml import etree
+from openupgradelib import openupgrade
+
+
+def prepare_dynamic_carousel(defaults_dict):
+    return """
+<section
+    data-snippet="s_dynamic_snippet"
+    class="s_dynamic_snippet_carousel s_dynamic pt24 pb24 o_colored_level d-none"
+    data-name="Dynamic Carousel"
+    data-number-of-elements="{products_per_slide}"
+    data-number-of-elements-small-devices="1"
+    data-carousel-interval="{interval}"
+    data-filter-id="{filter}"
+    data-number-of-records="{products_limit}"
+    data-template-key="website.dynamic_filter_template_image_title_footer"
+>
+    <div class="container o_not_editable">
+        <div class="css_non_editable_mode_hidden" />
+        <div class="dynamic_snippet_template"/>
+    </div>
+</section>
+    """.format(
+        **defaults_dict
+    )
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    if env["ir.module.module"].search(
```

Comment:
```suggestion
    if not env["ir.module.module"].search(
```
also, I think this will not work, as the module is merged to website_sale, so AFAIK it will not appear in ir.module.module

I think you have to do a different check to know that that module was installed. for example, check if website_sale.s_product_carousel xmlid exists.

Reply:
You're right. My initial approach was to do it in `pre-` but the I realized I needed some xlm_ids from 14.0. I'll try to guess it otherwise...

Reply:
Finally. I simply check if there are OCA's carousel snippets in the website views and change them when they're found.